external/storm-redis/src/main/java/org/apache/storm/redis/state/RedisKeyValueState.java

```diff
@@ -163,15 +190,20 @@ public void prepareCommit(long txid) {
                 LOG.debug("Prepared txn already exists, will merge", txid);
                 pendingPrepare.putAll(pendingCommit);
             }
-            if (!pendingPrepare.isEmpty()) {
+            if (!pendingPrepare.isEmpty() || !pendingDeletePrepare.isEmpty()) {
                 commands.hmset(prepareNamespace, pendingPrepare);
+                commands.hdel(prepareNamespace, pendingDeletePrepare.toArray(new String[pendingDeletePrepare.size()]));
```

Comment:
Yes, you are right, I forget easily that the `pendingPrepare`/`pendingCommit` doesn't actually contain the snapshot, just a delta